packages/gatsby-link/src/index.js

```diff
@@ -3,7 +3,7 @@ import { Link, NavLink } from "react-router-dom"
 import PropTypes from "prop-types"
 
 let pathPrefix = ``
-if (__PREFIX_PATHS__) {
+if (typeof __PREFIX_PATHS__ !== "undefined" && __PREFIX_PATHS__) {
```

Comment:
I think it would be better to use lodash for this check as it reads better and the code can be more consistent.
import {isUndefined} from 'lodash'
_.isUndefined(__PREFIX_PATHS__)